DynamicProgramming/KadaneAlgorithm.java

```diff
@@ -43,13 +44,16 @@ public static void main(String[] args) {
         Scanner sc = new Scanner(System.in);
         int n, arr[], i;
         n = sc.nextInt();
-        arr = new int[n];
-        for (i = 0; i < n; i++) {
-            arr[i] = sc.nextInt();
+        if(n>=0)
```

Comment:
Rather than failing silently when the user provides invalid input, why not add a branch that prints a helpful message (like "the input must be greater than or equal to zero") and then exits?

Reply:
Yeah, this is a much better way. I have made this change.